content/docs/15.how-to-guides/dataform.md

```diff
@@ -0,0 +1,202 @@
+
+### Using Dataform project creation in Kestra
+
+We will first create the Kestra flow and save it. The flow contains the following tasks:
+
+1. HTTP Download task that downloads the orders.csv file using HTTP URL.
+2. BigQuery CreateTable task that creates the `orders` table in `ecommerce` dataset.
+3. BoigQuery Load task that loads the `orders.csv` contents into BigQuery `orders` table.
+4. DataformCLI task that will run the Dataform project that we will create later using NamespaceFiles. The project creates the `stg_orders` BigQuery view based on the `orders` BigQuery table.
+
+```yaml
+id: dataform_project
+namespace: company.team
+
+tasks:
+  - id: orders_http_download
+    type: io.kestra.plugin.core.http.Download
+    description: Download orders.csv using HTTP Download
+    uri: https://huggingface.co/datasets/kestra/datasets/raw/#main/csv/orders.csv
+
+  - id: create_orders_table
+    type: io.kestra.plugin.gcp.bigquery.CreateTable
+    description: Create orders table in BigQuery
+    serviceAccount: "{{ secret('GCP_SERVICE_ACCOUNT_JSON') }}"
+    projectId: <gcp-project-id>
+    dataset: ecommerce
+    table: orders
+    tableDefinition:
+      type: TABLE
+      schema:
+        fields:
+        - name: order_id
+          type: INT64
+        - name: customer_name
+          type: STRING
+        - name: customer_email
+          type: STRING
+        - name: product_id
+          type: INT64
+        - name: price
+          type: FLOAT64
+        - name: quantity
+          type: INT64
+        - name: total
+          type: FLOAT64
+
+  - id: load_orders_table
+    type: io.kestra.plugin.gcp.bigquery.Load
+    description: Load orders table with data from orders.csv
+    from: "{{ outputs.orders_http_download.uri }}"
+    projectId: <gcp-project-id>
+    serviceAccount: "{{ secret('GCP_SERVICE_ACCOUNT_JSON') }}"
+    destinationTable: "<gcp-project-id>.ecommerce.orders"
+    format: CSV
+    csvOptions:
+      fieldDelimiter: ","
+      skipLeadingRows: 1
+
+  - id: dataform_cli
+    type: io.kestra.plugin.dataform.cli.DataformCLI
+    beforeCommands:
+      - npm install @dataform/core
+      - dataform compile
+    namespaceFiles:
+      enabled: true
+    env:
+      GOOGLE_APPLICATION_CREDENTIALS: "sa.json"
+    inputFiles:
+      sa.json: "{{ secret('GCP_SERVICE_ACCOUNT_JSON') }}"
+      .df-credentials.json: |
+        {
+          "projectId": "<gcp-project-id>",
+          "location": "us"
+        }
+    commands:
+      - dataform run
+```
+
+Once the flow is saved, navigate to the Editor, and create a file `package.json` with the contents:
+
+```json
+{
+    "dependencies": {
+        "@dataform/core": "2.3.0"
+    }
+}
+```
+
+Note that this file is not required for the Kestra run as we will be installing this dependency using `beforeCommands`. This is however required if you choose to push the namespace files to Github repository so that you can run the project seamlessly in other ways.
```

Comment:
Can you clarify the Kestra run part. Do you mean execution here?

```suggestion
Note that this file is not required for the Kestra run as we will be installing this dependency using `beforeCommands`. This is however required if you choose to push the namespace files to GitHub repository so that you can run the project seamlessly in other ways.
```